sdks/python/apache_beam/transforms/enrichment_handlers/vertex_ai_feature_store.py

```diff
@@ -0,0 +1,289 @@
+        Defaults to `ExceptionLevel.WARN`.
+      kwargs: Optional keyword arguments to configure the
+        `aiplatform.gapic.FeaturestoreOnlineServingServiceClient`. When using
+        `kwargs`, the `api_endpoint` param will be overridden with this config
+        if multiple values are found for `client_options`.
+    """
+    self.project = project
+    self.location = location
+    self.api_endpoint = api_endpoint
+    self.feature_store_id = feature_store_id
+    self.entity_type_id = entity_type_id
+    self.feature_ids = feature_ids
+    self.row_key = row_key
+    self.exception_level = exception_level
+    self.kwargs = kwargs if kwargs else {}
+
+  def __enter__(self):
+    """Connect with the Vertex AI Feature Store (Legacy)."""
+    try:
+      # checks if feature store exists
+      _ = aiplatform.Featurestore(
+          featurestore_name=self.feature_store_id,
+          project=self.project,
+          location=self.location,
+          credentials=self.kwargs.get('credentials'),
+      )
+      if 'client_options' in self.kwargs:
+        self.client = aiplatform.gapic.FeaturestoreOnlineServingServiceClient(
+            **self.kwargs)
+      else:
+        self.client = aiplatform.gapic.FeaturestoreOnlineServingServiceClient(
+            client_options={'api_endpoint': self.api_endpoint}, **self.kwargs)
+      self.entity_type_path = self.client.entity_type_path(
+          self.project,
+          self.location,
+          self.feature_store_id,
+          self.entity_type_id)
+    except NotFound:
+      raise ValueError(
+          'Vertex AI Feature Store %s does not exist' % self.feature_store_id)
+
+  def __call__(self, request: beam.Row, *args, **kwargs):
+    """Fetches feature value for an entity-id from
+    Vertex AI Feature Store (Legacy).
+
+    Args:
+      request: the input `beam.Row` to enrich.
+    """
+    try:
+      entity_id = request._asdict()[self.row_key]
+    except KeyError:
+      raise ValueError(
+          "no entry found for row_key %s in input row" % self.row_key)
+
+    try:
+      selector = aiplatform.gapic.FeatureSelector(
+          id_matcher=aiplatform.gapic.IdMatcher(ids=self.feature_ids))
+      response = self.client.read_feature_values(
+          request=aiplatform.gapic.ReadFeatureValuesRequest(
+              entity_type=self.entity_type_path,
+              entity_id=entity_id,
+              feature_selector=selector))
+    except NotFound:
+      raise ValueError(
+          _not_found_err_message(
+              self.feature_store_id, self.entity_type_id, entity_id))
+
+    response_dict = {}
+    proto_to_dict = proto.Message.to_dict(response.entity_view)
+    for key, msg in zip(response.header.feature_descriptors,
+                        proto_to_dict['data']):
+      if msg and 'value' in msg:
+        for _, value in msg['value'].items():
+          response_dict[key.id] = value
+          break  # skip fetching the metadata
```

Comment:
Can this just be `response_dict[key.id] = msg['value'].items()[0]`? Is this guaranteed to be the first element?

Reply:
Yeah, the first element is always value followed by metadata